src/v5.rs

```diff
@@ -192,20 +193,28 @@ impl Socks5Stream {
         let proxy_addr = read_response(&mut socket)?;
 
         Ok(Socks5Stream {
-            socket: socket,
-            proxy_addr: proxy_addr,
+            socket,
+            proxy_addr,
         })
     }
 
     fn password_authentication(socket: &mut TcpStream, username: &str, password: &str) -> io::Result<()> {
         if username.len() < 1 || username.len() > 255 {
-            return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid username"))
+            return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid username"));
         };
         if password.len() < 1 || password.len() > 255 {
-            return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid password"))
+            return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid password"));
         }
 
-        let mut packet = [0; 515];
+        // From RFC 1929:
+        // Request:
+        //  +----+------+----------+------+----------+
+        //  |VER | ULEN |  UNAME   | PLEN |  PASSWD  |
+        //  +----+------+----------+------+----------+
+        //  | 1  |  1   | 1 to 255 |  1   | 1 to 255 |
+        //  +----+------+----------+------+----------+
+        const PACKET_MAX: usize = 1 + 1 + 255 + 1 + 255;  // hopefully will optimized by compiler
```

Comment:
I really like the ASCII art tables but I'd probably remove the `// hopefully will optimized by compiler`. Since only constant expressions (=expression that can be evaluated at compile time) can be assigned to a `const`, it's a safe bet it will be evaluated at compile time. No hoping, praying, etc. necessary.

Reply:
Agreed